File: Lista3/zad2/Pieciakat.cpp
#include "Pieciakat.hpp"
#include <math.h>
#include <string>


double Pieciakat::obliczPole()
{
    return bok*bok/4*sqrt(25+10*sqrt(5));
}

double Pieciakat::obliczObwod()
{
    return bok*5;
}

Pieciakat::Pieciakat(double bok)
{
    this->bok = bok;
}

std::string Pieciakat::podajNazwe()
{
    return "Pieciakat";
}

File: Lista3/zad1/Pieciakat.java
public class Pieciakat extends Figura{
    double a;

    public Pieciakat(double a)
    {
        this.a = a;
    }

    public double obliczObwod() {
        return a*5;
    }
    
    public double obliczPole() {
        return a*a/4*Math.sqrt(25+10*Math.sqrt(5));
    }

    public String podajNazwe()
    {
        return "Pieciokat";
    }
}
